src/java/org/apache/cassandra/service/Systemd.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import org.apache.cassandra.utils.SigarLibrary;
+
+public class Systemd
+{
+    private static final Logger logger = LoggerFactory.getLogger(Systemd.class);
+
+    private static final Api INSTANCE = init();
+
+    private interface RawApi extends Library
+    {
+        int sd_pid_notify(int pid, int unset, String state);
+    }
+
+    public static class Api
+    {
+        private final RawApi impl;
+
+        private Api(RawApi rawApi)
+        {
+            this.impl = rawApi;
+        }
+
+        public void notifyReady()
+        {
+            long pid = SigarLibrary.instance.getPid();
```

Comment:
this needs to be rewritten as we do not use sigar anymore but oshi

Reply:
for cassandra builds on jdk11+, ProcessHandle.current().pid() is usable without an extra dependency